TCAT/Controllers/RouteOptionsViewController.swift

```diff
@@ -320,7 +321,65 @@ class RouteOptionsViewController: UIViewController {
         navigationItem.hidesBackButton = true
         searchBarView.searchController?.isActive = true
     }
-
+    
+    @objc func rerenderLiveTracking(sender: Timer) {
+        // Reload table every time update timer is fired
+        routeResults.reloadData()
+    }
+    
+    private func getDelay(tripId: String, stopId: String) -> Future<Response<Int?>> {
+        return networking(Endpoint.getDelay(tripID: tripId, stopID: stopId)).decode()
+    }
+
+    @objc func updateLiveTracking(sender: Timer) {
+        for routesArray in routes {
+            for (index, route) in routesArray.enumerated() {
+                if !route.isRawWalkingRoute(),
+                    let direction = route.getFirstDepartRawDirection(),
+                    let tripId = direction.tripIdentifiers?.first,
+                    let stopId = direction.stops.first?.id {
+                    getDelay(tripId: tripId, stopId: stopId).observe(with: { result in
+                        let fileName = "RouteTableViewCell"
+                        DispatchQueue.main.async {
+                            switch result {
+                            case .value (let delayResponse):
+                                guard (delayResponse.data != nil), let delay = delayResponse.data else {
+                                    return
+                                }
+                                // LUCY - Check
+                                let isNewDelayValue = (route.getFirstDepartRawDirection()?.delay != delay)
+                                if isNewDelayValue {
+                                    JSONFileManager.shared.logDelayParemeters(timestamp: Date(), stopId: stopId, tripId: tripId)
+                                    JSONFileManager.shared.logURL(timestamp: Date(), urlName: "Delay requestUrl", url: Endpoint.getDelayUrl(tripId: tripId, stopId: stopId))
+                                    if let data = try? JSONEncoder().encode(delayResponse) {
+                                        do { try JSONFileManager.shared.saveJSON(JSON.init(data: data), type: .delayJSON(rowNum: index)) } catch let error {
+                                            let line = "\(fileName) \(#function): \(error.localizedDescription)"
+                                            print(line)
+                                        }
+                                    }
+                                }
+                                let departTime = direction.startTime
+                                let delayedDepartTime = departTime.addingTimeInterval(TimeInterval(delay))
```

Comment:
might be wrong but from how i'm understanding it, as long as `delay > 0`, then we have a delay right? if so, we can simplify the logic a bit to:
```
if delay > 0 {
  let delayedDepartTime = departTime.addingTimeInterval(TimeInterval(delay))
  let delayState = DelayState.late(date: delayedDepartTime)
} else {
  let delayState = let delayState = DelayState.onTime(date: departTime)
}
```

Reply:
Looking at the code, it seems that this should be true but I'm doing some print statements and it seems like for small delay numbers (<35), we still consider that as on time even though its > 0? I believe this may have to do with the fact that our delay checks happen on a timer so some values may be a bit off. I think Omar would have to confirm though. @Omarrasheed

Reply:
ah yeah i think we do that because our "live tracking" isn't actually live tracking since TCAT only gives us their updated bus locations every 30 seconds